src/App.jsx

```diff
@@ -14,57 +15,70 @@ export const goods = [
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const handleSelect = item => {
+    setSelectedGood(item);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  const handleClear = () => {
+    setSelectedGood('');
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood ? `${selectedGood} is selected` : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={handleClear}
+          />
+        )}
+      </h1>
+      <table className="table">
+        <tbody>
+          {goods.map(item => (
+            <tr
+              key={item}
+              className={
+                selectedGood === item ? 'has-background-success-light' : ''
+              }
+              data-cy="Good"
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              <td>
+                {selectedGood !== item && (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={`button ${item}`}
+                    onClick={() => handleSelect(item)}
+                  >
+                    +
+                  </button>
+                )}
+                {selectedGood === item && (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className="button is-info"
+                    onClick={() => setSelectedGood('')}
+                  >
+                    -
+                  </button>
+                )}
```

Comment:
Here recommend using the ternary operator for conditional rendering, this will reduce the use of conditional constructions, improve code readability.
```suggestion
                {selectedGood !== item ? (
                  <button
                    data-cy="AddButton"
                    type="button"
                    className={`button ${item}`}
                    onClick={() => handleSelect(item)}
                  >
                    +
                  </button>
                ) : (
                  <button
                    data-cy="RemoveButton"
                    type="button"
                    className="button is-info"
                    onClick={() => setSelectedGood('')}
                  >
                    -
                  </button>
                )}
```